unified-runtime/source/loader/layers/sanitizer/tsan/tsan_libdevice.hpp

```diff
@@ -0,0 +1,145 @@
+/*
+ *
+ * Copyright (C) 2025 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+ * Exceptions. See LICENSE.TXT
+ *
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file tsan_libdevice.hpp
+ *
+ */
+
+#pragma once
+
+#include "sanitizer_common/sanitizer_libdevice.hpp"
+
+#if !defined(__SPIR__) && !defined(__SPIRV__)
+namespace ur_sanitizer_layer {
+#endif // !__SPIR__ && !__SPIRV__
+
+// Thread slot ID.
+typedef uint8_t Sid;
+
+// Abstract time unit, vector clock element.
+typedef int Epoch;
+
+const uintptr_t kThreadSlotCount = 256;
+
+// Count of shadow values in a shadow cell.
+const uintptr_t kShadowCnt = 2;
+
+// That many user bytes are mapped onto a single shadow cell.
+const uintptr_t kShadowCell = 8;
+
+// Single shadow value.
+typedef uint32_t RawShadow;
+const uint64_t kShadowSize = sizeof(RawShadow);
+
+// Shadow memory is kShadowMultiplier times larger than user memory.
+const uintptr_t kShadowMultiplier = kShadowSize * kShadowCnt / kShadowCell;
+
+typedef uint32_t AccessType;
+
+enum : AccessType {
+  kAccessWrite = 0,
+  kAccessRead = 1 << 0,
+  kAccessAtomic = 1 << 1,
+};
+
+// Fixed-size vector clock, used both for threads and sync objects.
+struct VectorClock {
+  Epoch clk_[kThreadSlotCount] = {};
+};
+
+struct Shadow {
```

Comment:
UR needn't know this structure? If so, please move to "tsan_rtl.hpp" or "tsan_rtl.cpp"

Reply:
done